### ISSUE TITLE
Umple template included uses current class context for further includes

### ISSUE THREAD

**Issue by Diusrex** [umple/umple#735]
For the setup:

```
class A
{
     template <<! From A !>>
     templateIncluded <<! <<@ template >> !>>
}

class B
{
    template <<! From B !>>
    reference <<! <<@ A.templateIncluded >> !>>

    emit genText()(reference);
}
```

genText will print out 'From B', which is not what would be expected since it would make sense for the `templateIncluded` to reference the `template` in class A.

This can be easily worked around by doing `A.template` instead, and may actually be desirable in some situations, by letting the user let a template be more customized, based on the class it is in.

However, if this configuration is not changed, then the code

```
class A
{
    similarTemplate <<! hi there !>>
    template <<! <<@similarTemplate >> !>>
}

class B
{
    similarTemplate <<! <<@ A.template >> !>>
    templateBFail <<! <<@similarTemplate>> !>>

    emit template3()(templateBFail);
}
```

will actually cause a StackOverflowError:

```
java.lang.StackOverflowError
        at java.util.regex.Pattern.expr(Pattern.java:1996)
        at java.util.regex.Pattern.compile(Pattern.java:1696)
        at java.util.regex.Pattern.<init>(Pattern.java:1351)
        at java.util.regex.Pattern.compile(Pattern.java:1028)
        at java.lang.String.replaceAll(String.java:2223)
        at cruise.umple.compiler.Utils.escapeString(Utils.java:215)
        at cruise.umple.compiler.Utils.nlize(Utils.java:131)
        at cruise.umple.compiler.TextElement.emit(TextElement.java:66)
        at cruise.umple.compiler.IncludeTemplateElement.emit(IncludeTemplateElement.java:67)
        ...
        at cruise.umple.compiler.IncludeTemplateElement.emit(IncludeTemplateElement.java:67)
```

since the cycle is not detected.

**Comment by TimLethbridge** [comment]
@ahmedvc Just merged pull request #738  that may solve this issued also #736 and #732 . Would @Diusrex comment and if the problem is solved, then close this issue please.

**Comment by Diusrex** [comment]
The issue of Umple not importing templates from the class a template was originally defined in, rather than the class that it is being imported into, has been fixed.
